Add lab 1

Zhavoronkova-Alina opened the pull request.

KotsMikhail committed:
Updated table

KotsMikhail committed:
Updated table

KotsMikhail committed:
Update table

KotsMikhail committed:
Updated table

KotsMikhail committed:
Updated table

KotsMikhail committed:
Updated table

KotsMikhail committed:
Updated table

KotsMikhail committed:
Updated table

KotsMikhail committed:
Updated table

KotsMikhail committed:
Update table

KotsMikhail committed:
Updated table

KotsMikhail committed:
Added deadlines

KotsMikhail committed:
Updated table

KotsMikhail committed:
Updated table

KotsMikhail committed:
Updated table

KotsMikhail committed:
Updated table

KotsMikhail committed:
Updated table

KotsMikhail committed:
Updated table

KotsMikhail committed:
Updated table

KotsMikhail committed:
Updated table

Zhavoronkova-Alina committed:
Add lab1

Create deamon for first lab

KotsMikhail commented on mitrofanova.alina/lab1/main.cpp:
```diff
@@ -0,0 +1,225 @@
+		ifs >> dir1 >> dir2;
+		ifs.close();
+	}
+	else {
```
Общепринятый стиль выглядит так: } else {

KotsMikhail commented on mitrofanova.alina/lab1/main.cpp:
```diff
@@ -0,0 +1,225 @@
+
+	std::ofstream ofs(hist_file, std::ios_base::out | std::ios_base::app); // open file for writing and append
+	if (!ofs.is_open()) {
+		syslog(LOG_INFO, "Don't open file");
```
"Don't open file" переводится как "Не открывать файл". Сомневаюсь, что это то, что имелось в виду

KotsMikhail commented on mitrofanova.alina/lab1/build.sh:
```diff
@@ -0,0 +1,12 @@
+
+chmod 666 "$pid_location"
+
+gcc -Wall -Werror main.cpp -o Lab1_Daemon -lstdc++
```
Если вы используете 11 стандарт С++ то при сборке следует указывать флаг -std=c++11, без него на некоторых системах и версиях компилятора собираться не будет

KotsMikhail reviewed (changes requested):
При попытке собрать вашу программу командой sudo ./build.sh получил следующую ошибку:
main.cpp: In function ‘void SetPidFile()’:
main.cpp:87:17: error: cannot resolve overloaded function ‘is_open’ based on conversion to type ‘bool’
if (ofs.is_open) {
В задании указано, что интервал должен быть в конфиге, у вас его нет

Zhavoronkova-Alina committed:
Update build.sh

Add C++11 standart

Zhavoronkova-Alina committed:
Update config.conf

Add interval to config file

Zhavoronkova-Alina committed:
Change code style if-else in main.cpp

Zhavoronkova-Alina committed:
Update main.cpp

- Add reading interval from config file
- Fix bug with is_open()
- Change the behaivor when hist.log could not be opened

Mitrofanova-Alina commented:
Исправила замечания. Жду ответа

KotsMikhail commented on mitrofanova.alina/lab1/main.cpp:
```diff
@@ -0,0 +1,219 @@
+	if (ifs.is_open()) {
+		ifs >> pid;
+		if (pid == -1) {
+			syslog(LOG_ERR, "Error pid and could not kill daemon");
```
Это сообщение об ошибке переводится как "Ошибка pid и не смог убить демона"

KotsMikhail commented on mitrofanova.alina/lab1/main.cpp:
```diff
@@ -0,0 +1,219 @@
+}
+
+char* CurrentTime() {
+	char* buffer = (char*)malloc(80);
```
Использовать рандомные константы в коде - плохой стиль, лучше объявлять их в начале программы

KotsMikhail commented on mitrofanova.alina/lab1/main.cpp:
```diff
@@ -0,0 +1,219 @@
+std::string dir1;
+std::string dir2;
+std::string hist_file = "hist.log";
+std::string pid_file = "/var/run/lab1_daemon.pid";
```
Путь к пид файлу должен быть константой

KotsMikhail reviewed (changes requested):
При посыле сигнала sighup программа всегда завершается с ошибкой о несуществующем конфиге
Программу можно запустить, не указывая в конфиге интервал, при этом демон будет работать с интервалом 0 и не выдаст никаких ошибок.

Zhavoronkova-Alina committed:
Separate constant declaration

Zhavoronkova-Alina committed:
Make constant path to pid-file

Zhavoronkova-Alina committed:
Make interval checking

Zhavoronkova-Alina committed:
Make some english improvement

Zhavoronkova-Alina committed:
Fix a signal SIGHUP error

Mitrofanova-Alina commented:
Исправила замечания.

KotsMikhail reviewed (approved):
Засчитано
